specification/cognitiveservices/data-plane/Face/preview/v1.0-preview/Face.json

```diff
@@ -4691,13 +4702,22 @@
           "type": "string",
           "maxLength": 64,
           "pattern": "^[a-z0-9-_]+$"
+        },
+        "name": {
+          "type": "string",
+          "description": "User defined name, maximum length is 128.",
+          "maxLength": 128,
+          "minLength": 1
+        },
+        "userData": {
+          "type": "string",
+          "description": "User specified data. Length should not exceed 16KB.",
+          "maxLength": 16384
+        },
```

Comment:
Is there a reason these aren't just references? That way you don't have multiple copies of the same thing, which would be a maintenance headache.

Reply:
As explained in the PR description, changing the name parameter from optional to required in the referenced structure caused AutoRest to generate methods with flipped parameter orders, causing a breaking change. Expanding it to have everything explicitly defined under each usage was unfortunately the only viable solution.

Reply:
I get why you have to break out the properties manually and can't just use "allOf", but I don't see why you need to repeat the same information three times in each swagger as opposed to something like this:

```
        "name": {
          "$ref": "#/definitions/UserDefinedName"
        },
        "userData": {
          "$ref": "#/definitions/UserData"
        },
```

Reply:
Got it. I did try that first, but the generated annotation parameter somehow discarded the description field. Not sure if it's a bug in AutoRest 2 but since we are sticking to that version for the time being, I had to copy/paste here to get them to show up.